### DISCUSSION TITLE
程序员如何学英语

### DISCUSSION THREAD

**Discussion by chen3feng** [chen3feng/article#77] [Ideas]
# 程序员如何学英语

由于教育方式等诸多方面的问题，很多人英语基础并不是很好，比如我大学四级考了5次才过。工作以后虽然一直没放弃，也只是限于搜索问题和阅读简短的工作用资料。我加大英语学习力度是最近几年的事，大致从2015年自助去美国旅游开始。几年下来各方面都提高了不少，信息渠道一下子扩大了好几倍，挺有成就感的。

## 首先，为什么学英语

英语是世界语言，事实上的“世界语”，就像普通话在中国的地位。不懂英语，你就处于一个信息孤岛中，就像穷乡僻壤的人在老家如鱼得水，到了大城市却格格不入一样，非常不方便。

- 工作需要
  作为一个程序员，最新的技术资料，最好的技术书籍，都是英文的，还有很多资料根本没有中文版本。遇到问题，google，stackoverflow，github等上的往往都有现成的答案，大幅度提高工作效率，提高技术水准。我想非技术岗位也类似。

- 出国活动
  出国旅游，外派，进修，英语都是必须的，如果定居，那更是必须的生活技能。

- 养育后代
  现在的小孩子学英语，大人也学好，日常也可以辅导甚至尝试某些时间进行全英语沟通，有助于共同提高。如果老了，小孩子移民了，跟过去也更容易适应。

- 扩大视野
  中文是世界上最大的信息孤岛，而且有大量的劣质信息。而英文世界的信息量是中文世界的十倍，即使劣质信息比例一样，优质信息的量也高很多。由于中国人整体英文水平较低，现在郭嘉对英文网站尚未大规模封杀，不用翻墙就能获得大量高实效性的优质信息。

- 提高自信
  看着英文好的朋友同事能和老外侃侃而谈，确实挺羡慕，希望自己也能拥有这种能力。

## 那么，如何学好英语？

学习英语需要投入一定的精力，作为一个有家有口的中年人，我的主要心得是，分配好时间，刻意练习，注意方法，贵在坚持。

手机不但是通讯和娱乐工具，也为我们学习提供了方便。装一些学英语的App，把一些看娱乐八卦，聊天灌水的时间抽出来学习英文，每天只需要半个小时到一个小时，日积月累，就会由量变到质变。

学习任何语言，最重要的是都听说读写，雅思考试什么的也是这么考的。如果你的目标是口语沟通，听说就比较重要。如果是要查资料，阅读能力比较重要，如果还要进行书面沟通，那么写也很重要。

### 词汇

词汇是非常重要的，英语为母语的人词汇量2万起，受过高等教育的往往能到3万以上。我们的大学4级词汇量只要求4千，6级6千，专业1万，而母语者小学入学就有一万的词汇量。也就是说，我们大部分受过高等教育的人，连母语小学生的词汇量都不如。

积累词汇有两种方式：阅读和背单词。对于英语环境的人来说，阅读就能满足词汇量增长的需要，但是非英语环境的话，背单词是必须的。阅读积累词汇和背单词并不矛盾，是可以而且应该结合在一起的。每天花点时间阅读英语资料，不认识的单词，查词典时如果认为是比较重要的，要记录起来加入生词表，最好词典本身就支持加入生词表。还有生活中遇到的中文单词有意识地看能否用英语表达，不能就查了记到背单词里。举个例子，开车时，汽车方面的方向盘，雨刷，后视镜，环岛，掉头，会不会说？不会就可以查了记下来背诵。

单纯的背单词没有意义，要有短语和例句，要能只听发音的背诵环节，要能按记忆曲线背诵。推荐App：知米背单词，欧路词典。通过这种方式，我现在对英文菜单和食品包装上的配料表词汇都基本掌握。有个词汇量测试网站test.qsbdc.com（轻松背单词），看着得分从一开始的四千涨到现在的9千，阅读英文不再吃力，感觉还是挺值得的。

学单词一定要注意，阅读词汇量，听力词汇量，口语词汇量，书面词汇量是不同的集合。相当于4个不同的hash表，单词进入了一个表，并不会自动扩散到别的表，需要全面练习。

- 阅读词汇量：就是阅读时见了知道意思的词汇，往往是最高的。人类通过视觉记忆的文字，就是单词的整体外形，很多非常用单词都只需要记住外形，不需要记住拼写，读音。积累的方式是多读。

- 听力词汇量：听了发音知道是什么词，是听力的关键，一般也能读出来，但是不一定会拼写。积累的方式是多听。

- 口语词汇量：表达时能轻松想到用起来，并且知道发音能念出来的词，一般都比较低。这个主要通过说来积累，如果有外语环境当然很好，在没有外语环境的情况下，可以和其他学习者进行英语沟通，以及尝试思考时用英语表达脑子里的想法。

- 书写词汇量：书写能想到会用，能写出来，一般是口语词汇量的子集。要通过多写来积累。在书写时，单词的固定搭配往往比单词本身更重要，就像类似汉语中的成语一样。为了阅读记忆的单词，到书写阶段就需要去整体记忆。

学单词还需要注意词频，不同的英语单词使用频率不同，多掌握高频单词，比掌握低频单词有价值的。因此有个比较快捷的方式是，找一个高频词表，从高频词开始，筛掉已经掌握的单词，把生词倒入背单词软件，背完再搞下一批，大概到8000就可以停止了，剩余的单词靠阅读来积累。

### 阅读

每天花几分钟读一些英语资料，养成可以看英文的习惯。工作中遇到英语不要回避，生活见到不认识的英语单词都查了记下来，比如食品包装，网传外媒截图原文，机场酒店医院的英语标牌等，推荐App：BBC News，无需翻墙。新闻类的用词和语法往往比较正式，社交网络，论坛类的往往更口语化，如果能翻墙，还可以看看推特什么的。

### 听力

练习听力的方式有精听和泛听。精听是必不可少的，每天精听十分钟，有空泛听，推荐App：每日英语听力，有文字和语音的对照。第一遍只听，第二遍对着文字听，第三遍再不看文字听，隔几天再重听。泛听我一般在开车时会进行，比如深圳地区的广播香港英文3台RTHK3 AM 567kHz

#### 关于口音

研究表明，母语对大脑的语言区有巨大的影响力，母语的发音和听力会塑造我们的大脑，导致我们区分不了外语中不同的音，不是，而且根本听不出区别，自然就说不出来，即使模仿口型能说出来，自己也听不出区别。

0～6岁是学语言的关键，这期间学习的语言叫习得，在长大就是学习，离不开母语的帮助，这是没有办法的事情。对于6岁以后才学外语的人来说，口音是不可避免的，但是只要想到你能轻松理解周围人南腔北调的汉语普通话，就能明白口音其实不是大问题，而且母语者对外国人的口音宽容度更高（人家能学会咱的语言，已经不容易了）。英语始终只是沟通工具，一开始不要给自己定太高的目标。

但是对于自己都能听出区别的音，还是能提高的，应该通过多听多练，能提高一些是一些。

## 总结

英语是很重要的沟通工具，希望大家都能掌握好，在更广阔的信息世界里自由翱翔。

**Comment by lisight** [comment]
写的很好! 特别是听说读写是四个表那段
